### PR TITLE
Added EasyMotion Highlighting.

### PR DESCRIPTION
**Pull request by hallzy**
Easy motion will now highlight single character targets with bold red, and the first character of a 2 character target will also be bold red, while the second will be bold purple.

If this colour scheme doesn't work for you obviously we can change it.

selection_071

### CODE REVIEW

**Comment by webdesus** [comment]
what proffit?

**Comment by hallzy** [comment]
Not certain I understand your question, but if you mean "Why bother", it is because easy-motion is not explicitly defined otherwise, and if you switch colour schemes while in vim then onedark doesn't highlight easy-motion at all (For example, when I switch from BadWolf (which has explicit highlighting for easy-motion) to OneDark I lose easy-motion highlighting). The letters just change silently in the file and it is impossible to see where:

Before starting to use easy-motion:
selection_072

After using easy-motion to look for the letter "o".
selection_073

This PR would make it look like this instead:
selection_074

**Comment by webdesus** [comment]
oh sorry...  i'm not correct understood you. I was think problem only with color. And i began think about (why 2 color better).  I think will be better make first letter red, second and others purple. So it will make more sense. I will understand where first letter, were second and another. And looking only red.

**Comment by joshdick** [comment]
For me, [easymotion/vim-easymotion](https://github.com/easymotion/vim-easymotion) works with onedark.vim without any additional configuration as of this writing: 

screen shot 2018-02-21 at 11 07 58 pm

As you wrote in your opening comment, you don't see *any* color when using easymotion without the code in your PR? It has default colors built in, according to `:help easymotion-custom-hl`.

Do you have other plugins that are manipulating those highlights? What version(s) of Vim have you seen this behavior with?

**Comment by hallzy** [comment]
I said in my last comment:

> easy-motion is not explicitly defined otherwise, and if you switch colour schemes while in vim then onedark doesn't highlight easy-motion at all

Because other colorschemes **DO** explicitly define easymotion colours and onedark does **NOT** when you switch colorschemes while inside of vim to OneDark then onedark will not show the highlighting.

When I start up vim using onedark right away and use easy motion I get what you get which is:

selection_075

I then switch to a different colorscheme (in this particular example, I switched to Iceberg). If I use easymotion now while using the iceberg colorscheme then this is what I get:

selection_076

I then switch back to OneDark and do an easymotion search for "O" again and this time I get this:

selection_078

Basically, this is an issue for those that change colorschemes or use different colorschemes for different files, or different types of files.

**Comment by joshdick** [comment]
Got it, that makes perfect sense.

Instead of directly merging this PR, I tweaked some whitespace for consistency and tweaked the colors so that they most match the easymotion-provided defaults, in https://github.com/joshdick/onedark.vim/commit/4f0292533a9960978e73c32245967b2efc66666b.

Thanks very much for the PR and for your detailed explanation of the issue.
